### PR TITLE
feat: support both enable/disable attribute for multi value headers on ALB

### PR DESCRIPTION
**Pull request by fredericgermain**
Settings in lambda target group.

*Issue #, if available:*

*Description of changes:*

# Checklist

- [ ] Tests have been added and are passing
- [ ] Documentation has been updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/event-sources/aws/alb.js [hunk 1]
```diff
@@ -2,16 +2,24 @@ const { getRequestValuesFromEvent, getMultiValueHeaders } = require('../utils')
...
   isBase64Encoded
 }) => {
-  const multiValueHeaders = getMultiValueHeaders({ headers })
+  const multiValueHeaders = !event.headers ? getMultiValueHeaders({ headers: responseHeaders }) : undefined
+  const headers = event.headers
+    ? Object.entries(responseHeaders).reduce((acc, [k, v]) => {
+      acc[k] = Array.isArray(v) ? v[0] : v
+      return acc
+    }, {})
+    : undefined
```
Comment (brettstack) [inline src/event-sources/aws/alb.js:17]: Could you add some comments on what we're trying to do here? Why are we using `event.headers` (request headers) for response headers?

**Comment by brettstack** [comment]
Thanks for the PR. Could you add more context on why this is needed? Thanks!

**Comment by fredericgermain** [comment]
When multi header value setting is disabled in target group (the default, in an ALB config), we receive this kind of event :

`{
  requestContext: {
    elb: {
      targetGroupArn: 'arn:aws:elasticloadbalancing:us-east-1:1234:targetgroup/my-cool-lambda/0fe13e0123456789'
    }
  },
  httpMethod: 'GET',
  path: '/',
  queryStringParameters: {
    data: '%7B%22type%22%3D%3E%22web%22%7D'
  },
  headers: {
    accept: '*/*',
    host: 'my-cool-lambda.my-awesome-domain.com',
    'user-agent': 'curl/7.68.0',
    'x-amzn-trace-id': 'Root=1-605c80b5-16f003190233a8156c949c15',
    'x-forwarded-for': '1.2.3.4',
    'x-forwarded-port': '443',
    'x-forwarded-proto': 'https'
  },
  body: '',
  isBase64Encoded: false
}
`

When multi value headers setting is enabled, we have this kind of event

`{
  requestContext: {
    elb: {
      targetGroupArn: 'arn:aws:elasticloadbalancing:us-east-1:1234:targetgroup/my-cool-lambda/0fe13e0123456789'
    }
  },
  httpMethod: 'GET',
  path: '/',
  multiValueQueryStringParameters: {
    data: [ '%7B%22type%22%3D%3E%22web%22%7D' ]
  },
  multiValueHeaders: {
    accept: [ '*/*' ],
    host: [ 'my-cool-lambda.my-awesome-domain.com' ],
    'user-agent': [ 'curl/7.68.0' ],
    'x-amzn-trace-id': [ 'Root=1-605c8726-68350c5341138f4d0e54f2e0' ],
    'x-forwarded-for': [ '1.2.3.4' ],
    'x-forwarded-port': [ '443' ],
    'x-forwarded-proto': [ 'https' ]
  },
  body: '',
  isBase64Encoded: false
} `

This patch accomodates both configurations dynamically.

**Comment by peebles** [comment]
Was this ever implemented?  I find that I *must* enable multi-value header option on ALB target group, or headers are not recognized at all in express server.  At least, I know for sure that CORs does not work at all unless multi-value headers are enabled.

**Review by brett-vendia — APPROVED** [review]

**Pull request merged**